var/spack/repos/builtin/packages/portage/package.py

```diff
@@ -17,14 +17,19 @@ class Portage(CMakePackage):
 
     # tarballs don't have submodules, so use git tags
     version('develop', branch='master', submodules=True)
+    version('1.2.2', tag='v1.2.2', submodules=True)
     version('1.1.1', tag='v1.1.1', submodules=True)
     version('1.1.0', tag='v1.1.0', submodules=True)
 
+    patch('gcc-7.patch', when='@1.2.2 %gcc@7:')
+    patch('p_lapacke_config.patch', when='@1.2:')
```

Comment:
Why are these patches needed?  Why weren't they needed before?  Can you point to webpages explaining these issues in greater detail?  Is the `gcc-7` patch needed for `v1.1.0` and `v1.1.1`?

Reply:
I haven't tested all combinations, I just don't want to touch old versions.

The `gcc-7.patch` is needed as `fabs()` needs `math.h` with gcc-7, this issue got fixed in the code.

The `p_lapacke_config.patch` got fixed here: https://github.com/laristra/cinch/commit/f87f848269fac25aa5b8d0bd5d9c9b2d2d6fb0ad

Reply:
> The `p_lapacke_config.patch` got fixed here: [laristra/cinch@f87f848](https://github.com/laristra/cinch/commit/f87f848269fac25aa5b8d0bd5d9c9b2d2d6fb0ad)

Please add that URL into the package as a comment so people know why the patch is there.



> The `gcc-7.patch` is needed as `fabs()` needs `math.h` with gcc-7, this issue got fixed in the code.

What do you mean by "got fixed in the code"?  In the GCC code or the `portage` code?  If the latter, which version was it fixed in?  Please
1. Add a comment like the above, explaining why the patch is needed (and again; pointing to a URL if possible)
2. Change the `when=` clause to be your best guess as to the versions that need it.  If you think this is a GCC 7 issue that would most likely affect previous versions of `portage` (even if you haven't tested them), then that would be `when=%gcc@7:`.
3. Do you think that `1.2.2` is the last release of `portage` that needs this patch?  If so, then you would use `when=@:1.2.2 %gcc7%`.

Reply:
Comments added.